_assets/ci/Jenkinsfile.tests

```
@@ -125,22 +130,30 @@ pipeline {
   } // post
 } // pipeline
 
-def getDefaultUnitTestCount() {
+def isTestNightlyJob() {
   if (env.JOB_BASE_NAME == 'tests-nightly') {
+    return true
+  } else {
+    return false
+  }
+}
```

What if we make those lambdas for brevity?
```suggestion
def isTestNightlyJob = { env.JOB_BASE_NAME == 'tests-nightly' ? true : false }
```
Same for other functions.

---

Maybe this would work even: `def isTestNightlyJob = { env.JOB_BASE_NAME == 'tests-nightly' }`

---

Hm closure didn't work for some reason:
```
java.lang.NoSuchMethodError: No such DSL method 'getDefaultTimeout' found among steps
```

```
timeout(time: getDefaultTimeout(), unit: 'MINUTES')
...
def getDefaultTimeout = { isTestNightlyJob() ? 5*60 : 40 }
```

---

Yes, it would, i was more showing it as an example for the non-boolean functions that return numbers.

And why it fails... weird. I can verify it works with just groovy:
```sh
 > cat lambda.groovy 
#!/usr/bin/env nix-shell
#!nix-shell -i groovy -p groovy

def testFunc = { "x" == "y" }
println(testFunc())

def testArgFunc = { x -> "string: ${x}" }
println(testArgFunc("wat"))

 > ./lambda.groovy 
false
string: wat
```

---

Yeah, I tried in `groovysh` too, but got an error in Jenkins.

I dindn't investigate and research why, just changed to functions and it worked again:

from
```
def getDefaultTimeout = { isTestNightlyJob() ? 5*60 : 40 }
```
to
```
def getDefaultTimeout() { isTestNightlyJob() ? 5*60 : 40 }
```

Takes the same space and same readability.
Do you think it's worth to research on closures failure?

---

Probably not. It might be something stupid tho, like order of operations.

I kinda like the functions being defined at the end tho.